benches/lua.rs

```diff
@@ -59,109 +57,118 @@ fn add_fields(c: &mut Criterion) {
                         r#"
                         hooks.process = """
                             function (event, emit)
-                                event['{}'] = '{}'
+                                event.log['{}'] = '{}'
+
+                                emit(event)
                             end
                         """
                         "#,
                         key, value
                     );
-                    transforms::lua::v2::Lua::new(&toml::from_str::<LuaConfig>(&config).unwrap())
-                        .unwrap()
+                    (
+                        Event::new_empty_log(),
+                        transforms::lua::v2::Lua::new(
+                            &toml::from_str::<LuaConfig>(&config).unwrap(),
+                        )
+                        .unwrap(),
+                    )
                 },
-                |mut transform| {
-                    for _ in 0..num_events {
-                        let event = Event::new_empty_log();
-                        let event = transform.transform(event).unwrap();
-                        assert_eq!(event.as_log()[key], value_bytes_v2);
-                    }
+                |(event, mut transform)| {
+                    let event = transform.transform(event).unwrap();
+                    assert_eq!(event.as_log()[key], value_bytes_v2);
                 },
             )
         })
-        .sample_size(10),
+        .throughput(Throughput::Elements(1)),
     );
 }
 
 fn field_filter(c: &mut Criterion) {
-    let num_events: usize = 100_000;
-
+    let num_events = 10;
```

Comment:
How would you feel about making this per-event?

Reply:
I did switch this to `iter_batched`, but I think this one still fits better running a small batch given we want to, sometimes, filter an event (10% of the time). I could augment it to be an individual event that is randomly filtered or not filtered, but I figured there could be some difference in execution time for a filtered event vs. an unfiltered event.

I could change this to be two benchmarks for the filter though, one where it always filters and one where it never filters.